text/0162-refactoring-support.md

```diff
@@ -0,0 +1,665 @@
+  properties before and after the deployment. In this case, you can remove the
+  unwanted mapping from the mapping file.
+
+We recommend that you always use the `confirm` option at development time. This
+way, you can see what changes the CLI is going to make, and decide whether they
+are correct. This gives you a chance to modify any incorrect mappings (by, for
+example, exporting the mapping, editing it, and importing it again).
+
+### Rollback
+
+After refactoring the stack, the CLI will proceed with the deployment
+(assuming that is your choice). If the deployment fails, and CloudFormation
+rolls it back, the CLI will execute a second refactor, in reverse, to bring the
+resources back to their original locations.
+
+If you don't want the CLI to perform the rollback refactor, you can use the
+`--no-rollback` flag, which also controls the rollback behavior of the
+deployment.
+
+### Programmatic access
+
+The same refactoring feature is also available in the CDK toolkit library:
+
+```typescript
+declare const toolkit: Toolkit;
+declare const cx: ICloudAssemblySource;
+
+// To execute possible refactors as part of the deploy operation:
+await toolkit.deploy(cx, {
+  refactoring: RefactoringMode.EXECUTE_AND_DEPLOY
+});
+
+// Or, if you just want to refactor the stacks:
+await toolkit.refactor(cx);
+```
+
+---
+
+
+Ticking the box below indicates that the public API of this RFC has been
+signed-off by the API bar raiser (the `status/api-approved` label was applied to
+the RFC pull request):
+
+```
+
+[ ] Signed-off by API Bar Raiser @rix0rrr
+
+```
+
+## Public FAQ
+
+### What are we launching today?
+
+A new developer experience for CDK users, that allows them to change the
+location of a construct (stack and/or logical ID) without causing resource
+replacement. This new experience is available in the CDK CLI `deploy`
+and `refactor` commands, as well as in the toolkit library.
+
+### Why should I use this feature?
+
+If you ever find yourself doing one of the following, you will benefit from
+stack refactoring support:
+
+- Renaming constructs, either intentionally or by mistake.
+- Moving constructs within the construct tree. This could be just for better
+  organization, or to create reusable components.
+- Moving constructs between different stacks.
+- Renaming stacks.
+- Upgrading dependencies on construct libraries.
+
+### What if I incorrectly refactor a stack?
+
+The CLI offers mechanisms to prevent accidental refactoring, such as dry runs,
+interactive mode, and importing and exporting of mapping files. Nevertheless,
+mistakes can happen. The first thing to be aware of is that refactors don't
+affect the resources themselves. The worst that can happen is you ending up with
+incorrect resource IDs in CloudFormation. The second thing is that refactors can
+be always be reverted. Simply edit the mapping file if you have one (or create
+one from scratch if you don't), and apply it to the affected stacks.
```

Comment:
Is this something we should do automatically? Generate a `refactor-rollback-mapping.json` ?